sympy/solvers/bivariate.py

```diff
@@ -199,17 +199,60 @@ def _solve_lambert(f, symbol, gens):
     nrhs, lhs = f.as_independent(symbol, as_Add=True)
     rhs = -nrhs
 
+    even_degrees = [i for i in degree_list(lhs) if i%2 == 0]
+    t = Dummy('t', **symbol.assumptions0)
+
     lamcheck = [tmp for tmp in gens
                 if (tmp.func in [exp, log] or
                 (tmp.is_Pow and symbol in tmp.exp.free_symbols))]
     if not lamcheck:
         raise NotImplementedError()
 
+    def solve_even_degree_branches(diff, symbol, t, gens):
+        """
+        Helper function to consider even degree branches of lambert
+        equation if exists.
+
+        For example:
+        Equation (a/x + exp(x/2)).diff(x) = 0 is represented
+        in logform as both of the below equations.
+        1. 2*log(x) + x/2 - log(a) = 0
+        2. 2*log(-x) + x/2 - log(a) = 0
+
+        Input: diff, symbol, t, gens
+        diff - Includes a dummy variable t which can be replaced
+        with +symbol and -symbol.
+
+        Output: Returns combined solution after considering all
+        even degree branches of original equation.
+        """
+        llhs1, llhs2 = map(lambda i: diff.xreplace({t: i}), (symbol, -symbol))
+        if llhs1 == llhs2:
+            return _solve_lambert(llhs1, symbol, gens)
+        else:
+            sol1, sol2 = map(lambda i: _solve_lambert(i, symbol, gens), (llhs1, llhs2))
+            return list(set(sol1 + sol2))
```

Comment:
It comes out after python debugging that this test in `test_ode` :
```python
>>>eq5 = 2*x**2*f(x) + f(x)**3 + (x*f(x)**2 - 2*x**3)*f(x).diff(x)
>>>dsolve(eq5, hint='1st_homogeneous_coeff_best')
Eq(f(x), exp(2*C1 + LambertW(-2*x**4*exp(-4*C1))/2)/x)  # if set() is used
# [Eq(f(x), exp(2*C1 + LambertW(-2*x**4*exp(-4*C1))/2)/x),
# Eq(f(x), exp(2*C1 + LambertW(-2*x**4*exp(-4*C1))/2)/x)] --> if sol1 +sol2
```
The two branches of the equation i.e `-log(-_x**2) + 2*log(_X0) + log(-2*_C1 + log(_X0) + log(_x))` and `2*log(-_X0) - log(-_x**2) + log(-2*_C1 + log(_X0) + log(_x))`. Solution of both the different branches is returned from _solve_lambert to be the same i.e `Eq(f(x), exp(2*C1 + LambertW(-2*x**4*exp(-4*C1))/2)/x)`.

Moreover I had also applied a check above that `if llhs1 == llhs2` then ony once the equation is solved.